### PR TITLE
SAK-49916 Lessons: remove oembed code that used shutdown YQL service

### PR DESCRIPTION
**Pull request by ottenhoff**

### CODE REVIEW

File: lessonbuilder/tool/src/webapp/js/jquery.oembed.js [hunk 1]
```diff
@@ -178,151 +185,91 @@
...
+
+                if (!embedProvider.nocache) {
+                    src += '&jqoemcache=' + rand(5);
+                }
+
+                if (embedProvider.apikey) {
+                    src = src.replace('_APIKEY_', settings.apikeys[embedProvider.name]);
+                }
+
+                var code = $('<' + tag + '/>').attr('src', src).attr('width', width)
```
Comment (ern) [inline lessonbuilder/tool/src/webapp/js/jquery.oembed.js:207]: use let vs var

Comment (ottenhoff) [reply lessonbuilder/tool/src/webapp/js/jquery.oembed.js:207]: nooooo, this is upstream code ....

File: lessonbuilder/tool/src/webapp/js/jquery.oembed.js [hunk 2]
```diff
@@ -178,151 +185,91 @@
...
+                    },
+                    error: settings.onError.call(container, externalUrl, embedProvider)
+                }, settings.ajaxOptions || {});
+                $.ajax(ajaxopts);
+            } else {
+                success({code: externalUrl.replace(embedProvider.templateRegex, embedProvider.template)}, externalUrl, container);
+            }
+        } else {
+
+            var requestUrl = getRequestUrl(embedProvider, externalUrl);
```
Comment (ern) [inline lessonbuilder/tool/src/webapp/js/jquery.oembed.js:247]: use const vs var

File: lessonbuilder/tool/src/webapp/js/jquery.oembed.js [hunk 3]
```diff
@@ -334,140 +281,166 @@
...
+        return oembedData.html;
     };
 
-    $.fn.oembed.getGenericCode = function(url, oembedData) {
-        var title = ((oembedData.title) && (oembedData.title !== null)) ? oembedData.title : url,
-            code = '<a href="' + url + '">' + title + '</a>';
-        if (oembedData.html) code += "<div>" + oembedData.html + "</div>";
+    $.fn.oembed.getGenericCode = function (url, oembedData) {
+        var title = ((oembedData.title) && (oembedData.title !== null)) ? oembedData.title : url;
+        var code = '<a href="' + url + '">' + title + '</a>';
```
Comment (ern) [inline lessonbuilder/tool/src/webapp/js/jquery.oembed.js:371]: use let vs var

File: lessonbuilder/tool/src/webapp/js/jquery.oembed.js [hunk 4]
```diff
@@ -505,368 +478,462 @@
...
+                        + '<p class="oembedall-updated-at">Last updated: ' + data.data.pushed_at + '</p></div></div>';
+                }
+            }),
+        new $.fn.oembed.OEmbedProvider("facebook", "rich", ["facebook.com"], null
+            , {templateRegex: /.*\/([^\/]+)\/([^\/]+).*/,
+                template: function (url) {
+                    // adding script directly to DOM to make sure that it is loaded correctly.
+                    if (!$.fn.oembed.facebokScriptHasBeenAdded) {
+                        $('<div id="fb-root"></div>').appendTo('body');
+                        var script = document.createElement('script');
```
Comment (ern) [inline lessonbuilder/tool/src/webapp/js/jquery.oembed.js:766]: use let vs var

File: lessonbuilder/tool/src/webapp/js/jquery.oembed.js [hunk 5]
```diff
@@ -505,368 +478,462 @@
...
+                        script.type = 'text/javascript';
+                        script.text = '(function(d, s, id) {var js, fjs = d.getElementsByTagName(s)[0];if (d.getElementById(id)) return;js = d.createElement(s); js.id = id;js.src = "//connect.facebook.net/en_US/sdk.js#xfbml=1&version=v2.0";fjs.parentNode.insertBefore(js, fjs);}(document, "script", "facebook-jssdk"));';
```
Comment (ern) [inline lessonbuilder/tool/src/webapp/js/jquery.oembed.js:768]: use let vs var

Comment (ern) [reply lessonbuilder/tool/src/webapp/js/jquery.oembed.js:768]: also this function could be better formatted vs a single line

File: lessonbuilder/tool/src/webapp/js/jquery.oembed.js (hunk 1 above)
Comment (ern) [reply lessonbuilder/tool/src/webapp/js/jquery.oembed.js:207]: ahh ok

**Pull request merged**